pandas/core/indexes/base.py

```diff
@@ -2320,6 +2320,11 @@ def symmetric_difference(self, other, result_name=None):
         except TypeError:
             pass
 
+        # On equal MultiIndexes the difference is empty. Therefore an empty
+        # MultiIndex is returned GH13490
+        if self.nlevels > 1 and len(the_diff) == 0:
+            return type(self)([[] for _ in range(self.nlevels)],
```

Comment:
Sorry I was confusing it with `self._shallow_copy_with_infer`. But still, if I call `self._shallow_copy` with the_diff being empty `from_tuples` gives me TypeError: Cannot infer number of levels from empty list